demos/ble/shadow_ble/shadow_demo_ble_transport.c

```diff
@@ -0,0 +1,1330 @@
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe.
+ */
+static uint16_t globalSubscribePacketIdentifier = 0U;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe ACK to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t globalUnsubscribePacketIdentifier = 0U;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe ACK to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t globalPublishPacketIdentifier = 0U;
+
+/**
+ * @brief The flag to indicate pubAck packet is received.
+ */
+static volatile bool pubAckReceived = false;
+
+/**
+ * @brief The flag to indicate the mqtt session changed.
+ */
+static bool mqttSessionEstablished = false;
+
+/**
+ * @brief Global entry time into the application to use as a reference timestamp
+ * in the #prvGetTimeMs function. #prvGetTimeMs will always return the difference
+ * between the current time and the global entry time. This will reduce the chances
+ * of overflow for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t ulGlobalEntryTimeMs;
+
+/**
+ * @brief The simulated device current power on state.
+ */
+static uint32_t ulCurrentPowerOnState = 0U;
+
+/**
+ * @brief The flag to indicate the device current power on state changed.
+ */
+static bool stateChanged = false;
+
+/**
+ * @brief When we send an update to the device shadow, and if we care about
+ * the response from cloud (accepted/rejected), remember the clientToken and
+ * use it to match with the response.
+ */
+static uint32_t ulClientToken = 0U;
+
+
+/**
+ * @brief The return status of prvUpdateDeltaHandler callback function.
+ */
+static int32_t lUpdateDeltaReturn = EXIT_SUCCESS;
+
+/**
+ * @brief The return status of prvUpdateAcceptedHandler callback function.
+ */
+static int32_t lUpdateAcceptedReturn = EXIT_SUCCESS;
+
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t prvGetTimeMs( void );
+
+/**
+ * @brief This example uses the MQTT library of the AWS IoT Device SDK for
+ * Embedded C. This is the prototype of the callback function defined by
+ * that library. It will be invoked whenever the MQTT library receives an
+ * incoming message.
+ *
+ * @param[in] pxMqttContext MQTT context pointer.
+ * @param[in] pxPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pxDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void prvEventCallback( MQTTContext_t * pxMqttContext,
+                              MQTTPacketInfo_t * pxPacketInfo,
+                              MQTTDeserializedInfo_t * pxDeserializedInfo );
+
+
+
+static void bleChannelCallback( IotBleDataTransferChannelEvent_t event,
```

Comment:
Add a declaration and docs for all the functions.. Also, if not already done, please run uncrustify.